### PR TITLE
feat: new types added to widgets

### PR DESCRIPTION
**Pull request by yuriassuncx**
### Description

This Pull Request adds new types to the `widgets.ts` file, enhancing the usability of Deco Widgets for developers. With these changes, users can directly import the types into their code, simplifying widget development and maintenance.

### Main Changes

- **Addition of new types with specific formatting:**
  - `TextArea`: Represents a text area.
  - `RichText`: Represents formatted text.
  - `ColorInput`: Represents a color input, with a default value of `#000000`.
  - `CSSCode`: Represents CSS code.
  - `TypescriptCode`: Represents TypeScript code.
  - `JSONCode`: Represents JSON code.
  
- **Along with the existing types:**
  - `ImageWidget`: Represents an image URI.
  - `HTMLWidget`: Represents an HTML string.
  - `VideoWidget`: Represents a video URI.
  - `SiteRoute`: Represents a dynamic route with options loaded from `website/loaders/options/routes.ts`.
  - `MapWidget`: Represents a map.

### Justification

Importing these new types significantly improves the usability of Deco Widgets by:
- Standardizing how different types of data are represented and manipulated.
- Facilitating code maintenance and expansion, allowing developers to use specific types with predefined formatting.
- Increasing code readability and clarity, helping new developers quickly understand how data is structured and used.

### Usage Example:

If the user wants to import any Widget present in the documentation, they can simply import it through `widgets.ts`:

Test tsx-camp-task-VisualStudioCode2024-07-1019-11-58-ezgif com-video-to-gif-converter

### Expected Benefits

- **Improved Developer Experience:** Directly importing types into the code simplifies developers' daily work.
- **Consistency and Maintenance:** Standardized types make the code easier to maintain and expand.
- **Greater Code Clarity:** Specific types make the code more readable and clear, aiding in quick understanding and onboarding of new developers.

### Checklist

- [x] Added new types to the `widgets.ts` file.
- [x] Updated documentation with usage examples for each new type.
- [x] Conducted tests to ensure the new types are functioning as expected.

---

If you need any changes or adjustments, feel free to let me know!

### CODE REVIEW

File: admin/widgets.ts [hunk 1]
```diff
@@ -23,3 +23,37 @@ export type SiteRoute = string;
...
+ * @format code
+ * @language typescript
+ */
+export type TypescriptCode = string;
+
+/**
+ * @format code
+ * @language json
+ */
+export type JSONCode = string;
```
Comment (mcandeia) [inline admin/widgets.ts:59]:
```suggestion

/**
 * @format textarea
 */
export type TextArea = string;

/**
 * @format rich-text
 */
export type RichText = string;

/**
 * @format color-input
 * @default #000000
 */
export type Color = string;

/**
 * @format code
 * @language css
 */
export type CSS = string;

/**
 * @format code
 * @language typescript
 */
export type TypeScript = string;

/**
 * @format code
 * @language json
 */
export type Json = string;
```

**Review by mcandeia — CHANGES_REQUESTED** [review]
I prefer just `CSS` `TypeScript` (S is uppercase), `Color`, etc.

If we are going to follow the file standard it will end with `Widget` which I think is not necessary. Lets avoid redundant things.

**Comment by mcandeia** [comment]
Awesome contribution @yuriassuncx <3

**Comment by yuriassuncx** [comment]
> Awesome contribution @yuriassuncx <3

thx candeia <3 
Is there any other type of widget that you find convenient to add? I saw that the date picker widget was recently launched

**Comment by mcandeia** [comment]
> Is there any other type of widget that you find convenient to add? I saw that the date picker widget was recently launched
Thats a good question, I'll also take the opportunity to kindly ask you to contribute in our docs page: https://deco.cx/docs/en/developing-capabilities/section-properties/widgets#65830370-000


Here are the respective MD files:

en: https://github.com/deco-sites/starting/blob/main/docs/developing-capabilities/section-properties/widgets/en.md
pt: https://github.com/deco-sites/starting/blob/main/docs/developing-capabilities/section-properties/widgets/pt.md


Could you please update the docs page with the new widgets you've created?

**Review by mcandeia — APPROVED** [review]

**Pull request merged**